### PR TITLE
chore(sass-lint): add sass linting

### LINKED ISSUE #3372: Add some Sass linting
Currently we have unofficial [**scss-lint**](https://github.com/brigade/scss-lint) support (see [/app/styles/.scss-lint.yml](https://github.com/mozilla/fxa-content-server/blob/master/app/styles/.scss-lint.yml) for current rules), although you have to run it manually (`$ scss-lint app/styles/**/*.scss`) and requires a Ruby gem.

It'd be nice if we could convert it to something pure-Node and possibly add it to our linting pipeline. I've used [**sass-lint**](https://github.com/sasstools/sass-lint) on other projects with some success.

Currently, running `$ scss-lint` against master gives us around 177 warnings/errors, so we probably need to clean up some Sass and tweak some rules.

### PR DESCRIPTION
**Pull request by TDA**
Add sass-lint and grunt-sass-lint to our workflow.

Fixes #3372
@pdehaan

### CODE REVIEW

**Comment by pdehaan** [comment]
1. Sass refactor looks solid.
2. Don't see a npm-shrinkwrap.json edit, but I can't remember if this repo shrinkwraps devDependencies.
3. Don't see anybody consuming the new `sasslint` Grunt task. Not sure if we should add it to [grunttasks/lint.js](https://github.com/mozilla/fxa-content-server/blob/master/grunttasks/lint.js#L13) so it gets triggered on every PR &mdash; to keep everybody honest.

**Comment by pdehaan** [comment]
Plus, make sure we remove unused [/app/styles/.scss-lint.yml](https://github.com/mozilla/fxa-content-server/blob/3c22f8e706504a9dab685657963e3339ad6089bf/app/styles/.scss-lint.yml) file.

**Comment by TDA** [comment]
@pdehaan Removed the unused file, and added the sasslint to lint tasks.

**Comment by pdehaan** [comment]
LGTM.
I'll let @vladikoff @shane-tomlinson @philbooth @vbudhram do the final r? and pull the 🔫 .

File: app/styles/modules/_avatar_spinner.scss [hunk 1]
```diff
@@ -1,4 +1,6 @@
 .avatar-spinner {
+  @extend %fill-avatar;
```
Comment (vladikoff) [inline app/styles/modules/_avatar_spinner.scss]: need to check this visually

File: app/styles/modules/_avatar.scss [hunk 2]
```diff
@@ -38,17 +38,17 @@
...
 
-  &.avatar-settings-view {    
+  &.avatar-settings-view {
     display: block;
     margin: 0 auto;
     position: relative;
 
-    &:hover{
-      box-shadow: 0px 0px 0px 1px #0095dd;
+    &:hover {
```
Comment (vladikoff) [inline app/styles/modules/_avatar.scss]: 👍

**Pull request merged**